salt-api/src/main/java/org/corpus_tools/salt/common/tokenizer/Tokenizer.java

```diff
@@ -584,26 +638,32 @@ private void setClitics(LanguageCode language) {
 			}
 
 			// attempt to separate proclitics
-			p = Pattern.compile("^(" + PClitic + ")(.+)$");
-			m = p.matcher(lstTokens.get(i));
-			if (m.find() && !PClitic.isEmpty()) {
-				lstTokens.remove(i);
-				lstTokens.add(i, m.group(2));
-				lstTokens.add(i, m.group(1));
-				continue; // proclitic has been removed, but next token must
-				// still be checked
+			if (getClitics(language)!= null && getClitics(language).getProclitic() != null) {
+			p = Pattern.compile("^" + getClitics(language).getProclitic() + "(.+)$");
+			String token = lstTokens.get(i);
+				m = p.matcher(token);
+				if (m.find() && (!getClitics(language).getProclitic().isEmpty())) {
+					lstTokens.remove(i);
+					lstTokens.add(i, m.group(2));
+					lstTokens.add(i, m.group(1));
+					continue; // proclitic has been removed, but next token must
+					// still be checked
+				}
 			}
 
 			// attempt to separate enclitics
-			p = Pattern.compile("(.+)(" + FClitic + ")$");
-			m = p.matcher(lstTokens.get(i));
-			if (m.find() && !FClitic.isEmpty()) {
-				lstTokens.remove(i);
-				lstTokens.add(i, m.group(2));
-				lstTokens.add(i, m.group(1));
-				i++; // next token is a known enclitic, skip it
-				continue; // advance to get past the enclitic
-			}
+				if (getClitics(language) != null && getClitics(language).getEnclitic() != null) {
+				p = Pattern.compile("(.+)" + getClitics(language).getEnclitic() + "$");
+				m = p.matcher(lstTokens.get(i));
+				if (m.find() && (!getClitics(language).getEnclitic().isEmpty())) {

```

Comment:
You use `getClitics(language).getEnclitic()` three times in this block. It would be better to cache the returned object instead of accessing the clitics map all the time.